src/env.c

```diff
@@ -1,6 +1,26 @@
 #include "rlang.h"
 
+
+bool r_is_unbound_value(SEXP x) {
+  return x == R_UnboundValue;
+}
+
 SEXP r_mut_env_parent(SEXP env, SEXP new_parent) {
```

Comment:
Alternatively, maybe we should just using a naming convention for the argument.

Calling them input/output parameters might be more clear than mutable? https://en.wikipedia.org/wiki/Parameter_(computer_programming)#Output_parameters